File: Java/InvertTree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
class Solution {
    public TreeNode invertTree(TreeNode root) {
        if(root == null) return root;
        TreeNode tmp = root.left;
        root.left = root.right;
        root.right = tmp;

        invertTree(root.left);
        invertTree(root.right);

        return root;
    }

    public TreeNode invertTreeIterative(TreeNode root) {
        if(root == null) return root;
        Queue<TreeNode> q = new LinkedList<>();
        q.offer(root);

        while(!q.isEmpty()) {
            TreeNode n = q.poll();
            TreeNode tmp = n.right;
            n.right = n.left;
            n.left = tmp;
            if(n.left != null) q.offer(n.left);
            if(n.right != null) q.offer(n.right);
        }
        return root;
    }
}

File: Go/invert_tree.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func invertTree(root *TreeNode) *TreeNode {
    if root == nil {
        return root
    }

    tmp := root.Right
    root.Right = root.Left
    root.Left = tmp

    invertTree(root.Left)
    invertTree(root.Right)

    return root
} 